frontend/pages/ManageControlsPage/Scripts/Scripts.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useContext, useRef, useState } from "react";
+import { useQuery } from "react-query";
+import { AxiosError } from "axios";
+
+import { AppContext } from "context/app";
+import { NotificationContext } from "context/notification";
+import scriptAPI, {
+  IScript,
+  IScriptsResponse,
+} from "services/entities/scripts";
+
+import CustomLink from "components/CustomLink";
+
+import PremiumFeatureMessage from "components/PremiumFeatureMessage";
+import ScriptListHeading from "./components/ScriptListHeading";
+import ScriptListItem from "./components/ScriptListItem";
+import DeleteScriptModal from "./components/DeleteScriptModal";
+import FileUploader from "../components/FileUploader";
+import UploadList from "../components/UploadList";
+
+const baseClass = "scripts";
+
+const Scripts = () => {
+  const { isPremiumTier } = useContext(AppContext);
+  const [showDeleteScriptModal, setShowDeleteScriptModal] = useState(false);
+
+  const selectedScript = useRef<IScript | null>(null);
+
+  const { data: scripts, isLoading, isError } = useQuery<
+    IScriptsResponse,
+    AxiosError,
+    IScript[]
+  >(["scripts"], () => scriptAPI.getScripts(), {
+    retry: false,
+    refetchOnWindowFocus: false,
+    select: (data) => data.scripts,
+  });
+
+  // The user is not a premium tier, so show the premium feature message.
+  if (!isPremiumTier) {
+    return (
+      <PremiumFeatureMessage
+        className={`${baseClass}__premium-feature-message`}
+      />
+    );
+  }
+
+  const onClickDelete = (script: IScript) => {
+    selectedScript.current = script;
+    setShowDeleteScriptModal(true);
+  };
+
+  const onCancelDelete = () => {
+    selectedScript.current = null;
+    setShowDeleteScriptModal(false);
+  };
+
+  const onDeleteScript = () => {
+    selectedScript.current = null;
+    setShowDeleteScriptModal(false);
+  };
+
+  return (
+    <div className={baseClass}>
+      <p className={`${baseClass}__description`}>
+        Upload scripts to change configuration and remediate issues on macOS
+        hosts. You can run scripts on individual hosts.{" "}
+        <CustomLink
+          text="Learn more"
+          url="https://fleetdm.com/docs/using-fleet/scripts"
+          newTab
+        />
+      </p>
+      {scripts && scripts.length !== 0 && (
+        <UploadList
```

Comment:
What do you think about renaming this component `FileList`? It took me a minute to realize that this was the list of uploaded files. I think I was reading "upload" as a verb in this context (although "file" can also be a noun or verb lol but maybe in the web context we don't tend to think of the verb meaning of file).

Reply:
yah im fine with changing the name. its really generic to handle rendering any type of list items.
but i can change it for clarity. mind if I do that in the next PR to not blow this one up?